lms/djangoapps/email_marketing/management/commands/remove_user_attribute.py

```diff
@@ -0,0 +1,54 @@
+"""
+Management command to remove provided user var(s) from sail-thru
+"""
+from django.core.management.base import BaseCommand
+from django.contrib.auth.models import User
+
+from email_marketing.tasks import update_sailthru_vars
+from util.query import use_read_replica_if_available
+
+
+class Command(BaseCommand):
+    """
+        This command removes custom attributes from user profiles on sailthru
+        Example usage:
+            $ ./manage.py lms remove_user_attribute --attribute=euid --chunk-size=2000
+        OR
+            $ ./manage.py lms remove_user_attribute --attribute=euid,grade
+    """
+
+    help = 'Removes a given user var(s) from sailthru'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--attribute',
+            type=str,
+            required=True,
+            help='attribute(s) to remove from custom vars from user profile'
+        )
+
+        parser.add_argument(
+            '--chunk-size',
+            type=int,
+            default=3000,
+            help='Maximum number of users to fetch in one db call')
+        pass
+
+    def handle(self, *args, **options):
+        """
+        It fetches data from database in chunks and rolls celery task for each record
+        """
+        chunk_size = options['chunk_size']
+        attribute = options['attribute']
+
+        query = User.objects.values('email')
+        queryset = use_read_replica_if_available(query)
+        count = queryset.count()
+
+        for start in range(0, count, chunk_size):
+            page = queryset[start: start+chunk_size].iterator()
```

Comment:
Using this style of `[start: end]` slicing will generate `LIMIT... OFFSET` queries, which will get progressively more expensive as you go through the queryset in sharkfin-fashion. Any `OFFSET` query has to examine all the rows that came before it. So if you have something saying "give me 10,000 records, offset by 10,000,000", the database has to run the queryset and examine 10,000,000 records before returning you the 10,000 after that:

```
mysql> explain select email from auth_user order by email limit 10000 offset 10000000 \G
*************************** 1. row ***************************
           id: 1
  select_type: SIMPLE
        table: auth_user
         type: index
possible_keys: NULL
          key: email
      key_len: 764
          ref: NULL
         rows: 10010000   <--- Database has to examine this many rows.
        Extra: Using index
1 row in set (0.00 sec)
```

Using just `iterator()` without slicing should get around this, but if you want to keep precise control over the batching behavior as you have here, you can track the last seen email from each batch and make follow-on queries that way. That way, you're constantly making just limit queries like: `select email from auth_user where email > '[last email from prev batch]' order by email limit [chunk size];`